### DISCUSSION TITLE
Should we build a MacOs client app?

### DISCUSSION THREAD

**Discussion by ebattistini** [icloud-photos-downloader/icloud_photos_downloader#488] [Ideas]
I've been battling with iCloud Photos for now 3 days and I've yet to accomplish both ways
- Downloading original full resolution photos from Photos.app 
- Downloading original full resolution via docker+icloudpd on my Synology NAS

One thing I've observed is that the download in photos.app is a background download process, so every time you do something on your Mac it will deprioritise the download process. So far in 3 days I have downloaded 2000 images from my 46K library... 

On the NAS I was able to download 16K images so far and I think it probably is the most reliable options so far. 

Now I believe there could be a great opportunity here to help users that are not technologically savvy but do want to download their library to put onto a hard drive 

Is anyone in this repo versed in client side apps? I'm a product designer and could create a simple UI for this to run as a standalone app? and maybe it could be sold on the App Store? or downloadable on gumroad?

Anyone?

**Comment by priithaamer** [comment]
Had a similar thought some time ago that native Mac app would make this backup solution more approachable to a whole lot more people. I'd be happy to work on this, especially if the product design part is covered 😄

**Comment by ebattistini** [comment]
I have not really thought about it but could we consider doing a menubar app with a simple onboarding where you define download location, and authentication credentials.
This would be the easiest approach ?

**Comment by ebattistini** [comment]
I have played around last night to have a quick idea of what it could be.
I still have to do the preference panel but those screens are what I had in mind


![Frame 4](https://user-images.githubusercontent.com/6288298/199936573-88d34ec7-78a0-4c7c-b1cd-01a853ab26f7.jpg)

![Frame 11](https://user-images.githubusercontent.com/6288298/199936598-df2d6c2e-fce2-44d0-8842-0f88967a9dec.jpg)
![Frame 10](https://user-images.githubusercontent.com/6288298/199936590-17c3c460-f894-4c3f-8a90-bc517ba45179.jpg)
![Frame 12](https://user-images.githubusercontent.com/6288298/199936601-c2f838db-dd35-4f74-96ed-a431fa9eae1e.jpg)

**Comment by AndreyNikiforov** [comment]
IIUC the proposal is to combine existing functionality of `icloudpd` with scheduling (that is available today as crontab) into one app with native UI. Sounds useful.

Any reason we limit that to MacOS and exclude Windows (and, possibly, even Linux, but I suspect the audience may be small for Linux UI)?

Apple's own iCloud app (on Windows) does not download images, so it covers different use case (== we are not competing).

**Reply by ebattistini** [reply]
if there are any Windows users that would want this! of course we should consider it

**Comment by rmbults** [comment]
Am I correct in assuming that this is still a command line only tool? Idea of native app or menu bar item has been abandonned?

**Reply by AndreyNikiforov** [reply]
I am not aware of any planned investments in developing standalone version with UI. How are you using icloudpd today and what improvement are you seeking from the version with UI?

**Reply by rmbults** [reply]
I don't use it now because I have no technical knowledge and don't feel comfortable using command line tools. I think many Mac users feel the same.
